### ISSUE TITLE
Test output needs improvement

### ISSUE THREAD

**Issue by bpinto** [bpinto/oh-my-fish#315]
```
'Testing a suite run'..Array index out of bounds
~/.oh-my-fish/plugins/fish-spec/expect.fish (line 10):       set -g expected  $argv[1..(math "$i - 1")]
                                                                                        ^
in function “expect”
	called on standard input
	with parameter list “--to-equal No tests found.”
in function “it_has_an_output_if_suite_is_blank”
	called on line 1 of file -
from sourcing file -
	called on line 52 of file /usr/share/fish/functions/eval.fish
in function “eval”
	called on line 94 of file ~/.oh-my-fish/plugins/fish-spec/spec.eval.fish
in function “spec.eval”
	called on line 52 of file ~/.oh-my-fish/plugins/fish-spec/spec.run.fish
	with parameter list “it_has_an_output_if_suite_is_blank --unit --depth 1 ”
in function “spec.run”
	called on line 121 of file ~/build/bpinto/oh-my-fish/plugins/fish-spec/spec/suite.spec.fish
	with parameter list “-v”
✘
Expected..
to equalNo tests found...'Testing Oh-My-Fish `list.erase` function...'
  'It should erase multiple elements.'..
  'It should erase multiple elements from multiple lists.'...
  'It should erase multiple elements with --from option.'..
  'It should erase an element from a list.'.
  'It should erase an element from multiple lists.'..
  'It should erase an element from multiple lists when only one has the element.'.
  'It should erase an element with --from option.'.'Testing bak plugin'..........✘
Expecteda/ a.20150114_021644.bak/
to equala.20150114_021644.bak/ a/....'Oh-My-Fish test spec...'
```

Ref: https://travis-ci.org/bpinto/oh-my-fish/builds/46936755

**Comment by bucaran** [comment]
Here is an idea. Let me try giving it a shot using msg instead of spec.log? What you say?

**Comment by bpinto** [comment]
I think it could be a good solution.

**Comment by bucaran** [comment]
Will work on it right away!

**Comment by bucaran** [comment]
Any preferences regarding UI?

I like this http://unitjs.com/assets/img/screen_console_unitjs_and_mocha_demo2.png

**Comment by bpinto** [comment]
I like that UI a lot, it's clean.

I don't know what happens when a test fails, but I think the error message should be displayed after all tests have run.

**Comment by bpinto** [comment]
Like this: https://raw.githubusercontent.com/metaskills/mocha-phantomjs/master/public/images/reporter_spec.gif

But I wouldn't write the error message like
```
1) fails 1
2) fails 2
```

I would follow the same style of successful tests.

**Comment by bucaran** [comment]
OK. Cool.

**Comment by bpinto** [comment]
Do you have any work in progress on this issue? Otherwise I'll start working on it.

**Comment by bucaran** [comment]
[This bug](https://github.com/fish-shell/fish-shell/pull/1892#issuecomment-69967427) was breaking `fish-spec`, but now that it is almost out of the way I am positive about moving forward with the framework. 

This means however we will need to incorporate the patched `eval` into the framework for some time. Any thoughts?

P.S: We still don't have

**Comment by bpinto** [comment]
I'd rather work around this bug then incorporating the patched `eval`. Either checking the fish version or executing a function that would check whether this bug was fixed or not in the running fish shell.

Regarding the implementation, we could not support multiple describes, as long as the things we do support are tested. My goal is to have a tested working framework instead of a breaking full-featured framework.

**Comment by bucaran** [comment]
I spent long hours trying to work around the bug and every atttempt ended up breaking somewhere else almost unexpectedly. One way is to not use `eval` in some calls, but that does not guarantee it will work.

Here is my recommendation. Let me finish my current batch of work. This version, which should be now bug free (no eval bug, and handles all describes correctly with the changes and fixed discussed until now) we will use to test the framework __reliably__.

After that, if we can find a way to patch it without my eval and the tests do not break, then sure, we can do away with the new eval.

**Comment by bpinto** [comment]
:+1:

**Comment by bucaran** [comment]
Thanks. 
[See](https://github.com/bucaran/fish-shell/blob/fix-eval/share/functions/eval.fish), it's just a simple fix.

**Comment by bpinto** [comment]
:+1:

**Comment by bpinto** [comment]
Fixed by #326
